engine/admin/resource_constraints.md

```diff
@@ -14,6 +14,42 @@ on when you should set such limits and the possible implications of setting them
 
 ## Memory
 
+### Understand the risks of running out of memory
+
+It is important not to allow a running container to consume too much of the
+host machine's memory. On Linux hosts, if the kernel detects that there is not
+enough memory to perform important system functions, it throws an `OOME`, or
+`Out Of Memory Exception`, and starts killing processes in order to free up
+memory. Any process is subject to killing, including Docker and other important
+applications. This can effectively bring the entire system down if the wrong
+process is killed.
+
+Docker attempts to mitigate these risks by adjusting the OOM priority on the
+Docker daemon so that it will be less likely to be killed than system processes
+The OOM priority on containers is not adjusted. This makes it more likely that
+an individual container will be killed than that the Docker daemon or other
+system processes will be killed.  You should not try to circumvent these
+safeguards by manually setting `--oom-score-adj` to an extreme negative number
+or setting  or `--oom-disable-kill` on a container or the Docker daemon. 
```

Comment:
the docker daemon doesn't have these flags (disable), only oom-weight (don't recall the exact flag name)